dot/rpc/http.go

```diff
@@ -53,15 +53,36 @@ type HTTPServerConfig struct {
 	TransactionQueueAPI modules.TransactionStateAPI
 	RPCAPI              modules.RPCAPI
 	SystemAPI           modules.SystemAPI
-	External            bool
+	RPC                 bool
+	RPCExternal         bool
+	RPCUnsafe           bool
+	RPCUnsafeExternal   bool
 	Host                string
 	RPCPort             uint32
 	WS                  bool
 	WSExternal          bool
+	WSUnsafe            bool
+	WSUnsafeExternal    bool
 	WSPort              uint32
 	Modules             []string
 }
 
+func (h HTTPServerConfig) RPCUnsafeEnabled() bool {
```

Comment:
Can we make `HTTPServerConfig` pointer? This will end up creating multiple copies.
```
func (h *HTTPServerConfig) RPCUnsafeEnabled() bool
```